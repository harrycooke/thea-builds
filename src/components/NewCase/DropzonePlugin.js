import React, { Component } from 'react';
import Dropzone from 'react-dropzone';
import { withStyles } from "@material-ui/core/styles";
import CloudDownloadIcon from "@material-ui/icons/CloudDownload";
import CloudUploadIcon from "@material-ui/icons/CloudUpload";
import {
  Grid,
  Typography,
  Button,
  ButtonGroup,
  Fab,
  Card,
  Paper,
  Divider,
  TextField,
  MenuItem,
  FormControlLabel,
  FormControl,
  Checkbox,
  Dialog,
  DialogActions,
  DialogTitle,
  Slide,
  Menu,
  Avatar,
  ListItemAvatar,
  ListItemText,
  CircularProgress
} from "@material-ui/core";
import AddAPhoto from '@material-ui/icons/AddAPhoto';


const styles = theme => ({
  container: {
    height: "40%",
    marginTop: "7%",
    // width: "20%"
  },
  mainText: {
    [theme.breakpoints.down('sm')]: {
      fontSize: 15,
    },
    [theme.breakpoints.up('sm') ]: {
      fontSize: 20,
    },
    textAlign: "center",
    color: "#818181",
  }
});
const acceptedFileTypes = "application/*, image/*, video/mp4,video/x-m4v,video/*, text/*, audio/*, document/*, .rtf, .dot"
class DropzonePlugin extends Component {

  onDrop = (file) => {
    this.props.handleFileUpload(file)
  }

  // removeFile = (fileIndex) => {
  //   this.props.removeFile(fileIndex)
  // }

  render() {
    const { classes } = this.props;

    return (
              <ButtonGroup fullWidth variant="outlined" size="large" color="primary" orientation="vertical" aria-label="contained primary button group">
          <Button className={classes.container}>
            {this.props.isLoading && (
              <CircularProgress
                size={20}
                disableShrink
              />
            )}
            <div className={classes.dropzone} style={{ height: "100%", cursor: "pointer" }}>
              <div style={{
                display: "flex",
                justifyContent: "center",
                height: "100%",
                flexDirection: "column",
                alignItems: "center"
              }}>
                <AddAPhoto style={{ width: 80, height: 80 }} />
                <p  className={classes.mainText}>Grab Page From Open Application</p>
              </div>
            </div>

          </Button>

                     {/* <Button variant="outlined" style={{ paddingBottom: 50, height: "0%", minHeight: 100 }} aria-controls="simple-menu" aria-haspopup="true" onClick={this.handleMenuClick}>
                        Grab Page From Open Application
                      </Button>*/}

          <Button className={classes.container}>

      <Dropzone accept={acceptedFileTypes} onDrop={this.onDrop}>
        {({ getRootProps, getInputProps }) => (
          <div>
            {this.props.isLoading && (
              <CircularProgress
                size={20}
                disableShrink
              />
            )}
            <div {...getRootProps({ className: 'dropzone' })} style={{ height: "100%", cursor: "pointer" }}>
              <input {...getInputProps()} />
              <div style={{
                display: "flex",
                justifyContent: "center",
                height: "100%",
                flexDirection: "column",
                alignItems: "center"
              }}>
                <CloudUploadIcon style={{ width: 80, height: 80 }} />
                <p className={classes.mainText}>Drag and drop or click here to upload files</p>
              </div>
            </div>

          </div>

        )}
      </Dropzone>
      </Button>

      </ButtonGroup>
    );
  }
}

export default withStyles(styles)(DropzonePlugin);