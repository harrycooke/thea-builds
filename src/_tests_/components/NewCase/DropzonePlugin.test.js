import React from 'react';
import ReactDOM from'react-dom';

import TestRenderer from 'react-test-renderer'; // ES6\
import sinon from 'sinon';
import { mount } from 'enzyme';

import {DropzonePlugin} from '../../../components/NewCase/DropzonePlugin';

describe('DropzonePlugin', () => {
    beforeAll = () => {
        this.props = {};
    };

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(DropzonePlugin, div);
    });

    it('renders React components, traversing a ReactElement tree and represents as JSON object', () => {
        const renderer = TestRenderer.create(DropzonePlugin);
        renderer.toJSON();
      });
      
    describe('DropzonePlugin', () => {
        it('should render correctly in "debug" mode', () => {
          const component = (DropzonePlugin);
          expect(component).toMatchSnapshot();
        });
      });

    // it('calls funcName handleSnackbarClose', () => {
    //     sinon.spy(DropzonePlugin.prototype, 'handleSnackbarClose');
    //     const wrapper = mount(<DropzonePlugin />);
    //     expect(DropzonePlugin.prototype.componentDidMount).to.have.property('callCount', 1);
    //   });
    
    //   it('allows us to set props', () => {
    //     const wrapper = mount(<DropzonePlugin bar="baz" />);
    //     expect(wrapper.props().bar).to.equal('baz');
    //     wrapper.setProps({ bar: 'foo' });
    //     expect(wrapper.props().bar).to.equal('foo');
    //   });
    
    //   it('simulates click events', () => {
    //     const onButtonClick = sinon.spy();
    //     const wrapper = mount((
    //       <DropzonePlugin onButtonClick={onButtonClick} />
    //     ));
    //     wrapper.find('button').simulate('click');
    //     expect(onButtonClick).to.have.property('callCount', 1);
    //   });
});