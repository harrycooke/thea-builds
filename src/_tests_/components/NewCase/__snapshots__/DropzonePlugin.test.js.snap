// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropzonePlugin DropzonePlugin should render correctly in "debug" mode 1`] = `undefined`;
